tests/next_tests/unit_tests/embedded_tests/test_nd_array_field.py

```diff
@@ -632,3 +632,59 @@ def test_setitem_wrong_domain():
 
     with pytest.raises(ValueError, match=r"Incompatible 'Domain'.*"):
         field[(1, slice(None))] = value_incompatible
+
+
+def test_connectivity_field_inverse_image_implementation():
+    V = Dimension("V")
+    E = Dimension("E")
+
+    V_START, V_STOP = 2, 7
+    E_START, E_STOP = 0, 10
+
+    e2v_conn = common._connectivity(
+        np.roll(np.arange(E_START, E_STOP), 1),
+        domain=common.Domain(
+            dims=(E,),
+            ranges=[
+                UnitRange(E_START, E_STOP),
+            ],
+        ),
+        codomain=V,
+    )
+
+    # Test range
+    image_range = UnitRange(V_START, V_STOP)
+    result = e2v_conn.inverse_image(image_range)
+
+    assert len(result) == 1
+    assert result[0] == (E, UnitRange(V_START + 1, V_STOP + 1))
+
+    # Test cache
+    cached_result = e2v_conn.inverse_image(image_range)
```

Comment:
why is this testing the cache?

Reply:
This is testing the cache because it returns exactly the same result object as before. If the cache were not used, it would return a new object with exactly the same content.

Reply:
oh, I read the `is` as `==`